### ISSUE TITLE
Archive task failing with ENOBUFS

### ISSUE THREAD

**Issue by johnwc** [microsoft/azure-pipelines-agent#3991]
## Agent Version and Platform
Version of your agent? 2.211.0

OS of the machine running the agent? RHEL 8

## Azure DevOps Type and Version
dev.azure.com

## What's not working?
On-prem agent started getting the below error on archive task.
##[error]Error: Archive creation failed for archive file: /var/devops/agent1/_work/25/a/uat.zip 
```
stderr:  
error: Error: spawnSync /bin/zip ENOBUFS
    at Object.spawnSync (node:internal/child_process:1111:20)
    at Object.spawnSync (node:child_process:814:24)
    at ToolRunner.execSync (/var/devops/agent1/_work/_tasks/ArchiveFiles_d8b84976-e99a-4b86-b885-4849694435b0/2.211.0/node_modules/azure-pipelines-task-lib/toolrunner.js:879:23)
    at zipArchive (/var/devops/agent1/_work/_tasks/ArchiveFiles_d8b84976-e99a-4b86-b885-4849694435b0/2.211.0/archivefiles.js:152:33)
    at createArchive (/var/devops/agent1/_work/_tasks/ArchiveFiles_d8b84976-e99a-4b86-b885-4849694435b0/2.211.0/archivefiles.js:278:13)
    at doWork (/var/devops/agent1/_work/_tasks/ArchiveFiles_d8b84976-e99a-4b86-b885-4849694435b0/2.211.0/archivefiles.js:338:9)
    at Object.<anonymous> (/var/devops/agent1/_work/_tasks/ArchiveFiles_d8b84976-e99a-4b86-b885-4849694435b0/2.211.0/archivefiles.js:347:1)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32) {
  errno: -105,
  code: 'ENOBUFS',
  syscall: 'spawnSync /bin/zip',
  path: '/bin/zip',
  spawnargs: [Array]
```

**Comment by aagoldingay** [comment]
Exact same stack trace encountered on Ubuntu 20.04.5 on 2.211.0

**Comment by johnwc** [comment]
Verified also fails in azure hosted `ubuntu-latest` agents with same stack.

**Comment by adovy** [comment]
Extract files fails as well with ENOBUFS

**Comment by Bene-Graham** [comment]
Extract File task still fails with ENOBUFS error. The workaround of executing bash to extract the file worked.

**Comment by johnwc** [comment]
The correct workaround is to change `- task: ArchiveFiles@2` to `- task: ArchiveFiles@2.206.0` in your pipeline, forcing the pipeline agent to use the previous version. Though, I do not think this is needed as of today, as I believe they rolled out a fix for this already.

**Comment by DenisRumyantsev** [comment]
Fixed in `2.211.1`
